/**: # Common parameters for all nodes

  frames:
    robot_base: "base_link"
    odom: "map"
    world: "map"
    scan: "LDS-01"

mapper:
  loop_rate: 50.0

drive_manager:
  loop_rate: 5.0
  outputs:
    robot_command:
      name: 'cmd_vel'
      msg_type:   'Twist'

controller:
  # loop_rate: 10.0
  closed_loop: True
  loop_rate: 5.0
  control_time_step: 0.1
  prediction_horizon: 4.0

  Stanley:
    cross_track_gain: 1.0
    heading_gain: 2.0

  DVZ:
    cross_track_gain: 1.0
    heading_gain: 2.0
    K_angular: 1.0
    K_linear: 1.0
    min_front_margin: 1.0
    side_margin_width_ratio: 1.0

  DWA:
    control_horizon: 0.3
    octree_resolution: 0.1
    max_linear_samples: 20
    max_angular_samples: 20
    costs_weights:
      goal_distance_weight: 2.0
      reference_path_distance_weight: 1.5
      obstacles_distance_weight: 0.0
      smoothness_weight: 0.0
      jerk_weight: 0.0

planner:
  inputs:
    map_layer:
      name: '/map'
      msg_type:   'OccupancyGrid'
    goal_point:
      name: '/clicked_point'
      msg_type:  'PointStamped'

  loop_rate: 1.0
  algorithm: 'ompl'
  ompl:
    log_level: 'WARN'
    planning_timeout: 10.0
    planner_id: 'ompl.geometric.KPIECE1'
    KPIECE1:
      optimization_objective: 'MaximizeMinClearanceObjective'
